wire-ios/Wire-iOS/Sources/UserInterface/DeviceView/DeviceInfoViewModel.swift

```diff
@@ -171,11 +172,11 @@ final class DeviceInfoViewModel: ObservableObject {
         }
     }
 
-    func updateVerifiedStatus(_ value: Bool) async {
-        // TODO: Check why this is not working as expected
-        let result = await actionsHandler.updateVerified(value)
-        DispatchQueue.main.async {
-            self.isProteusVerificationEnabled = result
+    func updateVerifiedStatus(_ value: Bool) {
+        actionsHandler.updateVerified(value) {
+            DispatchQueue.main.async {
+                self.isProteusVerificationEnabled = self.userClient.verified
+            }
         }
```

Comment:
Shouldn't the final value be passed as parameter in the completion closure? it's kind of implicit knowledge here that `self.userClient.verified` was the changed result and needs to be used now.

```suggestion
        actionsHandler.updateVerified(value) { isVerified in
            DispatchQueue.main.async {
                self.isProteusVerificationEnabled = isVerified
            }
        }
```

Reply:
Here, `isProteusVerificationEnabled` needs to always display the verification status of `UserClient`. If not this would represent a wrong state when the result could say its that its verified but the `UserClient`'s status never changed.

Reply:
Then I would rather suggest to have a func like and you can keep the `async` logic:
```swift
@MainActor
func updateProteusVerification() {
    isProteusVerificationEnabled = userClient.verified
}
```

Reply:
Works now with the async changes.